src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1342,6 +1278,27 @@ static bool processContextAttributeVector (ContextElement*
 
     }
 
+    /* Special processing in the case of DELETE, to avoid "deleting too much" in the case of using
+     * metadata ID. If metadata ID feature were removed this ugly piece of code wouldn't be needed ;)
+     * FIXME P3: note that in the case of Active/Active configuration it could happen that
+     * the attrsName get desyncrhonized, e.g. starting situation is A-ID1 and A-ID2, node
+     * #1 process DELETE A-ID1 and node #2 process DELETE A-ID2, each one thinking than the
+     * other A-IDx copy will be there at the end, thus nobody includes it in the toPull array
+     */
+    if (strcasecmp(action.c_str(), "delete") == 0)
```

Comment:
I tried to find a method to do *strcasecmp* for std::string.
Found this: http://stackoverflow.com/questions/11635/case-insensitive-string-comparison-in-c

C++, the language of the future :-D

NTC